### PR TITLE
 	blockstore: fix PutMany with cache logic

### PR DESCRIPTION
**Pull request by Kubuxu**
Thanks @whyrusleeping for noticing it.

Removed PutMany logic in bloom cache as it can't help with anything.
Fixed ARC cache to use filtered results instead of all blocks.

Also add test case for it.

### CODE REVIEW

**Comment by whyrusleeping** [comment]
your tests are broken and stuff

**Comment by Kubuxu** [comment]
Hmm, interesting, they've run fine locally.

File: blocks/blockstore/bloom_cache.go [hunk 1]
```diff
@@ -126,19 +127,11 @@ func (b *bloomcache) Put(bl blocks.Block) error {
...
-		for _, block := range bs {
-			b.bloom.AddTS([]byte(block.Key()))
-		}
-	}
-	return err
+	// bloom cache gives only conclusive resulty if key is not contained
+	// to reduce number of puts we need conclusive infomration if block is contained
+	// this means that PutMany can't be improved with bloom cache so we just
+	// just do a passthrough.
+	return b.blockstore.PutMany(bs)
```
Comment (Kubuxu) [inline blocks/blockstore/bloom_cache.go]: I have to add all block to bloom....

Will add a test case.

**Comment by Kubuxu** [comment]
@whyrusleeping should be fixed, travis is queued up and previously it was having network problems.

**Comment by whyrusleeping** [comment]
LGTM, thanks for fixing :)

**Pull request merged**